controllers/argocd/deployment.go

```diff
@@ -626,12 +635,17 @@ func (r *ReconcileArgoCD) reconcileRedisDeployment(cr *argoprojv1a1.ArgoCD) erro
 // reconcileRedisHAProxyDeployment will ensure the Deployment resource is present for the Redis HA Proxy component.
 func (r *ReconcileArgoCD) reconcileRedisHAProxyDeployment(cr *argoprojv1a1.ArgoCD) error {
 	deploy := newDeploymentWithSuffix("redis-ha-haproxy", "redis", cr)
+	nodePlacementDeploy := newDeploymentWithSuffix("redis-ha-haproxy", "redis", cr)
```

Comment:
I think if there is no existing `redisHAProxyDeployment`, the new one is created without the `NodeSelector and Tolerations` since we are using the `deploy` object rather than `nodePlacementDeploy`.
I would recommend using only a single object i.e `deploy` and creating a new object `existing` to fetch the existing deployment. This would maintain consistency with other resources.

Reply:
Thanks a lot!!! will look at this right now!